lib/ansible/plugins/inventory/azure_rm.py

```diff
@@ -511,12 +527,38 @@ def hostvars(self):
                     new_hostvars['private_ipv4_addresses'].append(private_ip)
                 pip_id = ipc['properties'].get('publicIPAddress', {}).get('id')
                 if pip_id:
+                    new_hostvars['public_ip_id'] = pip_id
+
                     pip = nic.public_ips[pip_id]
+                    new_hostvars['public_ip_name'] = pip._pip_model['name']
                     new_hostvars['public_ipv4_addresses'].append(pip._pip_model['properties'].get('ipAddress', None))
                     pip_fqdn = pip._pip_model['properties'].get('dnsSettings', {}).get('fqdn')
                     if pip_fqdn:
                         new_hostvars['public_dns_hostnames'].append(pip_fqdn)
 
+            new_hostvars['mac_address'] = nic._nic_model['properties'].get('macAddress') if nic._nic_model['properties'].get('macAddress') else None
```

Comment:
you can just do:
new_hostvars['mac_address'] = nic._nic_model['properties'].get('macAddress')
because get() returns None by default if key doesn't exist